### ISSUE TITLE
Write SVG in PostCSS

### ISSUE THREAD

**Issue by ai** [postcss/postcss#553]
@ben-eb suggested great idea to write SVG inside CSS with some plugin like `postcss-write-svg`:

```css
.arrow {
    @svg {
        polygon {
            fill: green;
            points: 50,100 0,0 0,100; 
        }
    }
}
```

to:

```css
.arrow {
  background: url(data:…);
}
```

Of course big SVG like icons should be draw in vector editor. But small cases like “HTML art” for arrows and simple poligons can be done in CSS by hands.

**Comment by davidkpiano** [comment]
This would be fun to do in Sass as well... hmm...

**Comment by ai** [comment]
Yeap, you can write PostCSS plugin and then connect it after Sass (as Autoprefixer).

**Comment by Grawl** [comment]
👍

**Comment by pascalduez** [comment]
I was working on such a plugin, but ad to pause it while waiting for async PostCSS, now I should really revive it!
https://github.com/pascalduez/postcss-svg

I have uncommitted code to be able to manipulate svg's.

**Comment by MattDiMu** [comment]
@davidkpiano I did a similar plugin for sass&gulp, which is much simpler though. It's not about creating, but only about manipulating SVGs in sass. It just reads a given folder of SVG-files & creates url-encoded (for best gzip-encoding) SASS-functions, which take colors as parameters and return the manipulated SVG. Furthermore it offers some utiliy-functions for these SASS-functions. It started as an experiment, but I've already used it in 4 professional project and it seems to work with all single-color SVGs I've come across until now (icomoon, font awesome and some self-created SVGs).
https://github.com/MattDiMu/gulp-sassvg
